File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/wrapper/FruitWrapper.java
package com.devwilly.recyclerview_java.wrapper;

/**
 * Created by Willy on 15/01/2018.
 */

public class FruitWrapper implements IEatingType {

    private String mName;
    private String mPrice;

    public FruitWrapper(String name, String price) {
        this.mName = name;
        this.mPrice = price;
    }

    @Override
    public String getName() {
        return mName;
    }

    @Override
    public String getPrice() {
        return mPrice;
    }

    @Override
    public int getItemType() {
        return IEatingType.FRUIT;
    }
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/wrapper/FruitWrapper.kt
package com.devwilly.recyclerviewkotlin.wrapper

/**
 * Created by Willy on 15/01/2018.
 */
class FruitWrapper(
        private val name: String,
        private val description: String,
        private val price: String) : IEatingWrapper {

    override fun getName(): String {
        return name
    }

    override fun getDescription(): String {
        return description
    }

    override fun getPrice(): String {
        return price
    }

    override fun getItemType(): Int {
        return IEatingWrapper.FRUIT
    }
}